pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -193,7 +198,7 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
 
     // Validate data access
     AccessControl accessControl = _accessControlFactory.create();
-    if (!accessControl.hasDataAccess(httpHeaders, rawTableName)) {
+    if (!accessControl.hasAccess(rawTableName, AccessType.READ, httpHeaders, endpointUrl)) {
```

Comment:
Let's not change this. Although we're reading data and AccessType.READ looks appropriate, we need to only allow ppl who are authorized to be able to query the data that might contain sensitive info like user information. It's different than reading table config or some other cluster configs that might be ok with weaker authentication level like AccessType.READ.

Reply:
Similar to the above, this has similar strictness to the original implementation. The difference here is removing the use of a deprecated method.

On a tangent, this resource shouldn't be in the controller in the first place, rather, it should be controlled by the broker's access control.